### ISSUE TITLE
Utilize MemoryCache instead of a custom cache implementation?

### ISSUE THREAD

**Issue by intricate** [blockchyp/blockchyp-csharp#5]
It seems that `TerminalRouteCache` could perhaps benefit from composing with or inheriting from .NET's [`System.Runtime.Caching.MemoryCache`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.caching.memorycache?view=netframework-4.5). For C#/.NET devs this might expose a more familiar interface.

It's also worth looking at the source code for `MemoryCache`'s internal [`MemoryCacheStore`](https://referencesource.microsoft.com/#System.Runtime.Caching/System/Caching/MemoryCacheStore.cs). I'm not sure whether it's worth reimplementing much of this functionality when it seems that you'd just like to expand upon it (e.g. utilizing a cache located on the filesystem). 

But perhaps I'm missing the bigger picture and, for some reason, `MemoryCache` isn't a good fit?

Thoughts?

**Comment by intricate** [comment]
This is `MemoryCache`'s internal `MemoryCacheStore`: https://referencesource.microsoft.com/#System.Runtime.Caching/System/Caching/MemoryCacheStore.cs

I'm not sure whether you should reimplement much of this functionality when it seems that you'd just like to expand upon it (e.g. utilizing a cache located on the filesystem). 

But perhaps I'm missing the bigger picture and `MemoryCache` isn't a good fit.

**Comment by jniedrauer** [comment]
I think that this is overly complicated for our needs here. The route cache is internal and there's no reason developers using the SDK should ever need to interact with it. It's low throughput enough that performance and memory will never be limiting factors (famous last words). MemoryCache also doesn't support persistence as you noted, so it would require custom code to serialize and deserialize data to disk anyway.

**Comment by intricate** [comment]
No problem with this being closed and I'm happy to take your point here :smile:. 

However, just for the sake of keeping notes, I'll provide some more context as to why I suggested `MemoryCache` or something alike it.

> The route cache is internal and there's no reason developers using the SDK should ever need to interact with it.

I suppose I assumed that this wasn't necessarily the case. In the ["route cache" section of the BlockChyp SDK developer's guide](https://docs.blockchyp.com/sdk-guide/index.html#route-cache) it says:

> You may elect to provide a cache timeout... In the event of a cache timeout, consider waiting to see if your terminal route request succeeds before treating it as a cache miss.

I believe the Java SDK supports cache entry expiration so I figured that the current route cache implementation would probably be extended to support it as well (which is also supported by `MemoryCache` via [`CacheItemPolicy`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.caching.cacheitempolicy?view=netframework-4.5)). Because of this, I also figured that the entry expiration policy would be optionally be specified by the user of the SDK rather than being strictly hardcoded. However, I believe that the expiration time is actually hardcoded to 1 hour in the Java SDK.

However, I think I'd agree that it's fine to leave it as is and perhaps expand upon the implementation if/when necessary.
